### ISSUE TITLE
Method must be pre-JITted before dissasembly or else it will fail with System.InvalidOperationException: Unable to disassemble method

### ISSUE THREAD

**Issue by ghost** [xoofx/JitBuddy#2]
Consider this example:

```csharp
using System;
using System.Reflection;
using System.Runtime.CompilerServices;
using JitBuddy;

static class Program
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    static int GetNumber1(int number)
    {
        return number * 1;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static int GetNumber2(int number)
    {
        return number * 2;
    }

    static void Main()
    {
        //
        // force jit
        //
        // Console.WriteLine(GetNumber1(1));  // Uncomment to succeed
        // Console.WriteLine(GetNumber2(1));  // Uncomment to succeed

        //
        // then disasm
        //
        MethodInfo method1 = typeof(Program).GetMethod("GetNumber1", BindingFlags.Static | BindingFlags.NonPublic);

        Console.WriteLine(method1);
        Console.WriteLine(method1.ToAsm());

        MethodInfo method2 = typeof(Program).GetMethod("GetNumber2", BindingFlags.Static | BindingFlags.NonPublic);

        Console.WriteLine(method2);
        Console.WriteLine(method2.ToAsm());
    }
}
```

without first uncommenting lines marked with `Uncomment to succeed` comments, **second** `ToAsm()` will fail with:

```
Unhandled Exception: System.InvalidOperationException: Unable to disassemble method `Int32 GetNumber2(Int32)`
   at JitBuddy.JitBuddy.ToAsm(MethodInfo method, StringBuilder builder, Formatter formatter)
   at JitBuddy.JitBuddy.ToAsm(MethodInfo method)
```
(on .NET Core, x64, Release build)

or will emit empty string on .NET Framework 4.7.2, x64, Release build.

**Comment by ghost** [comment]
I don't think this is a bug in JitBuddy. It seems to be a feature of .NET that allows reflection without actually jitting methods. Users should be aware of this.

**Comment by xoofx** [comment]
Yeah, I will have to look at it, not sure it is related to this or it is another issue in the JitBuddy actually...

The code that is forcing JIT prepare on methods should be performed:

https://github.com/xoofx/JitBuddy/blob/d9336f5f89e9732558bf92ab8a350f26198eec6e/src/JitBuddy/JitBuddy.cs#L55

It is not going to work well with tiered compilation (so tiered compilation will have to be disabled), see issue https://github.com/dotnet/coreclr/issues/22123

**Comment by ghost** [comment]
I've disabled tiered compilation (`<TieredCompilation>false</TieredCompilation>`), but exception is still thrown.
Also, as far as I'm aware, .NET Framework does not have tiered compilation and it fails to disassemble too.
Might be entirely different issue, as you've said.

**Comment by xoofx** [comment]
Fix should hopefully be fixed by commit a532f29587c1c649a97e9feefd7cbd487a171dd1 and available in `0.2.0`

**Comment by ghost** [comment]
@xoofx  Excellent! Works as advertised. Thank you.
